pkg/migrate/cstor/backup_restore.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migrate
+
+import (
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	openebsio "github.com/openebs/api/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/api/pkg/apis/types"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TranslateBackupToV1 translates v1alpha resources to v1
+func TranslateBackupToV1(olBackup openebsio.CStorBackup) *cstor.CStorBackup {
+	newBackup := &cstor.CStorBackup{}
+	newBackup.ObjectMeta = metav1.ObjectMeta{
+		Name:        olBackup.Name,
+		Namespace:   olBackup.Namespace,
+		Labels:      olBackup.Labels,
+		Annotations: olBackup.Annotations,
+	}
```

Comment:
This can be simply `newBackup.ObjectMeta = olBackup.ObjectMeta` as objectMeta is separately versioned.

Reply:
Similarly, please check at other places.

Reply:
If we do that ObjectMeta will have the old `uid` which will result in Create failures.

Reply:
> If we do that ObjectMeta will have the old uid, creationTimeStamp, etc which will result in Create failures.

It will get overwritten by apiserver